### DISCUSSION TITLE
Disable merge-confidence for private artifacts

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#20279] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.100.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

In a self hosted Renovate runner on GitLab I have:
> RENOVATE_EXTENDS: 'github>whitesource/merge-confidence:beta'

(and also:
>   RENOVATE_ONBOARDING_CONFIG: '{"$$schema": "https://docs.renovatebot.com/renovate-schema.json", extends: ["config:base"], labels: ["deps"]}'

)

to have the merge confidence badges.

However, it doesn't make sense for private artifacts, so I would like to have it selectively disabled.

### What you tried so far.

In the global Renovate configuration I have tried:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  packageRules: [
    {
      matchPackagePrefixes: ["org.mockito"],
      ignorePresets: ["github>whitesource/merge-confidence:beta", ":prBodyDefinitions"],
      fetchReleaseNotes: false
    },
    ...
}
```

(and particular `ignorePresets` separately)

While release notes are disabled, I still have the merge confidence table :-/. How to fix it?

Btw, I was not able to find `config:base` "file" in the Renovate repository. Where is its definition specified?

**Comment by rarkins** [comment]
`ignorePresets` doesn't work inside `packageRules` - must be top level.

Instead, You should create your own merge confidence preset where you can put the settings within packageRules which exclude your own prefixes. The source is here: https://github.com/whitesource/merge-confidence/blob/main/beta.json
